mlflow_yarn/yarn_backend.py

```diff
@@ -0,0 +1,139 @@
+import logging
+import os
+import tempfile
+import time
+
+import conda_pack
+import skein
+import cluster_pack
+from cluster_pack.skein import skein_config_builder, skein_helper
+from cluster_pack import packaging
+
+
+from mlflow.entities import RunStatus
+from mlflow.projects.utils import fetch_and_validate_project, get_or_create_run
+from mlflow.projects.backend.abstract_backend import AbstractBackend
+from mlflow.projects.submitted_run import SubmittedRun
+from mlflow.projects import _get_or_create_conda_env, _get_entry_point_command, load_project
+from mlflow.exceptions import ExecutionException
+
+
+_logger = logging.getLogger(__name__)
+
+
+class YarnSubmittedRun(SubmittedRun):
+    """Instance of SubmittedRun
+       corresponding to a Yarn Job launched through skein to run an MLflow
+       project.
+
+    :param skein_app_id: ID of the submitted Skein Application.
+    :param mlflow_run_id: ID of the MLflow project run.
+    """
+    def __init__(self, client, skein_app_id, mlflow_run_id):
+        super(YarnSubmittedRun, self).__init__()
+        self._skein_client = client
+        self._skein_app_id = skein_app_id
+        self._mlflow_run_id = mlflow_run_id
+
+    @property
+    def run_id(self):
+        return self._mlflow_run_id
+
+    def wait(self):
+        result = skein_helper.wait_for_finished(self._skein_client, self._skein_app_id)
+        self._skein_client.close()

```

Comment:
I will not managed the skein client lifecycle here.

For instance, if the skein client is shared by several runs, it should be close outside from here.

Reply:
Removed the close()